### PR TITLE
Call setup() on autonomous modes, fixing #19

### PR DESCRIPTION
**Pull request by Noskcaj19**
This allows accessing magic injected variables from autonomous modes during initialization

### CODE REVIEW

**Comment by auscompgeek** [comment]
Hm. I think teams have been using `on_enable` to work around this.

**Comment by Noskcaj19** [comment]
That can work, however that is called when the mode is enabled, not when the robot starts.  If you have somewhat expensive operations, that may not be entirely practical.

File: magicbot/magicrobot.py [hunk 1]
```diff
@@ -498,6 +498,12 @@ def _create_components(self):
             if hasattr(component, 'setup'):
                 component.setup()
 
+        # Call setup functions for autonomous modes
+        for mode in self._automodes.modes.values():
+            if hasattr(mode, 'setup'):
+                mode.setup()
+
+
```
Comment (auscompgeek) [inline magicbot/magicrobot.py]: What's up with the extra blank line here?

**Comment by virtuald** [comment]
I suppose this is fine. I would prefer to fix #21 instead, really.

**Comment by virtuald** [comment]
But yes, please add some documentation in the right place and this is fine.

**Comment by Noskcaj19** [comment]
Sorry for the delay, as for the documentation, is there anywhere specific it should be added? Perhaps in robotpy-docs magicbot.rst in autonomous mode. If so, should I open a pr in that repo?

**Comment by auscompgeek** [comment]
Huh, I thought there was like a stub class documenting the interface somewhere here. I guess it is all in the robotpy-docs repo.

**Pull request merged**